pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -80,7 +80,7 @@ type Manager interface {
 
 	// GetCPUs implements the podresources.CPUsProvider interface to provide allocated
 	// cpus for the container
-	GetCPUs(podUID, containerName string) []int64
+	GetCPUs(podUID, containerName string) *cpuset.CPUSet
```

Comment:
This should not be a pointer to a CPUSet, but just a CPUSet.
Under the hood it is just a struct wrapping a map, so in terms of storage, it is already just passing a pointer around.

Reply:
I think the only overhead will be because of struct copy, why do you not like to use a pointer here?

Reply:
Mostly just because the `CPUSet` type was designed so that you wouldn't have to pass around a pointer to it.

It's already passed around efficiently as the CPUSet type itself. Passing a pointer to it is actually *less*  efficient, because the exact same size object is being passed around, but there is one more level of indirection you have to go through get to the underlying map.

In other words, the definiton of the `CPUSet` struct looks as follows:
```
type CPUSet struct {
    elems map[int]struct{}
}
```

Which (in terms of storage) is essentially the same as the following, since go stores `map`s as pointers:
```
type CPUSet struct {
    elems uintptr
}
```

Which then (in terms of storage) is exactly the same as a standalone `uintptr`.

So by passing a full `CPUSet` type, you are essentially just passing a direct copy of the underlying `uintptr` to its embedded map (with 0 extra overhead). By passing a pointer to a `CPUSet` you are still passing a `uintptr` around, but you have to follow one more level of indirection before getting to the underlying map's pointer.

Reply:
For reference:
https://play.golang.org/p/Eb0u32gTgDR

Reply:
I see thanks for the explanation.

Reply:
For reference:
https://play.golang.org/p/WeCYE-5MY9O

Reply:
For reference:
https://play.golang.org/p/TlclpQYkBLS